internal/daemonset/daemonset.go

```diff
@@ -162,8 +166,28 @@ func (dc *daemonSetGenerator) SetDriverContainerAsDesired(ctx context.Context, d
 		},
 	}
 
-	return dc.constructDaemonSet(ctx, ds, &mod, mod.Spec.DriverContainer, "driver-container", image, standardLabels,
```

Comment:
why not just use constructDaemonSet and pass it the finalizers?

Reply:
That's yet another parameter for a limited value, IMHO. And I guess the usefulness of that function will even decrease when we refactor the CRD.

Reply:
it just seems stange that we have it for device plugin and not for driver-container. essentially it is almost the same function flow

Reply:
My plan is to remove that call for the device plugin creation as well. The two `SetAsDesired` functions do similar work indeed, but calling a common function with many parameters makes the code less straightforward IMHO.

Reply:
Removed the `constructDaemonSet` method.